### PR TITLE
feat(ts): add missing locale type definitions

### PR DESCRIPTION
**Pull request by JounQin**
@iamkun This PR improves locale user experience for TypeScript users, please help to review.

### CODE REVIEW

**Comment by JounQin** [comment]
@iamkun Any time to review this?

File: types/locale/types.d.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+declare interface ILocale {
+  name: string
```
Comment (iamkun) [inline types/locale/types.d.ts]: name is required and others are not

File: types/index.d.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/// <reference path="./locale/index.d.ts" />
```
Comment (iamkun) [inline types/index.d.ts]: Not sure if we need this reference, can IDE get the path on its own, like our other plugins' types.

Comment (JounQin) [reply types/index.d.ts]: No, There is no `locale/*.d.ts` files except `index` and `type`, but other plugins have their own `.d.ts`.

File: types/locale/types.d.ts (hunk 1 above)
Comment (JounQin) [reply types/locale/types.d.ts]: @iamkun Done.

**Comment by JounQin** [comment]
@iamkun Any further idea? 🤣

**Comment by JounQin** [comment]
？？？@iamkun What's going on? Can you please take care of this?

**Comment by iamkun** [comment]
We will merge all the incoming features before the next release,  thanks for your concern.

**Comment by JounQin** [comment]
I saw you're merging other PRs during these days, so I was thinking is there anything wrong with this PR.

**Comment by iamkun** [comment]
Oh, I see, the PRs get merged quickly are all chore fix.

**Comment by JounQin** [comment]
Well, follow your own schedule then. Cheers. 🍺

**Comment by JounQin** [comment]
@iamkun Sorry I have to asking for releasing it again here, it is really needed...

**Comment by iamkun** [comment]
@JounQin let's make it merged today!

**Comment by JounQin** [comment]
@iamkun Thx!

**Pull request merged**